lib/modules/manager/npm/post-update/rules.ts

```diff
@@ -3,51 +3,83 @@ import * as hostRules from '../../../../util/host-rules';
 import { regEx } from '../../../../util/regex';
 import { toBase64 } from '../../../../util/string';
 import { isHttpUrl } from '../../../../util/url';
+import type { YarnRcYmlFile } from './types';
 
 export interface HostRulesResult {
   additionalNpmrcContent: string[];
   additionalYarnRcYml?: any;
 }
 
 export function processHostRules(): HostRulesResult {
-  let additionalYarnRcYml: any;
+  const additionalYarnRcYml: YarnRcYmlFile = { npmRegistries: {} };
 
   // Determine the additional npmrc content to add based on host rules
   const additionalNpmrcContent = [];
   const npmHostRules = hostRules.findAll({
     hostType: 'npm',
   });
   for (const hostRule of npmHostRules) {
-    if (hostRule.resolvedHost) {
-      let uri = hostRule.matchHost;
-      uri =
-        is.string(uri) && isHttpUrl(uri)
-          ? uri.replace(regEx(/^https?:/), '')
-          : // TODO: types (#22198)
-            `//${uri}/`;
-      if (hostRule.token) {
-        const key = hostRule.authType === 'Basic' ? '_auth' : '_authToken';
-        additionalNpmrcContent.push(`${uri}:${key}=${hostRule.token}`);
-        additionalYarnRcYml ||= { npmRegistries: {} };
-        if (hostRule.authType === 'Basic') {
-          additionalYarnRcYml.npmRegistries[uri] = {
-            npmAuthIdent: hostRule.token,
-          };
-        } else {
-          additionalYarnRcYml.npmRegistries[uri] = {
-            npmAuthToken: hostRule.token,
-          };
-        }
-      } else if (is.string(hostRule.username) && is.string(hostRule.password)) {
-        const password = toBase64(hostRule.password);
-        additionalNpmrcContent.push(`${uri}:username=${hostRule.username}`);
-        additionalNpmrcContent.push(`${uri}:_password=${password}`);
-        additionalYarnRcYml ||= { npmRegistries: {} };
-        additionalYarnRcYml.npmRegistries[uri] = {
-          npmAuthIdent: `${hostRule.username}:${hostRule.password}`,
+    if (!hostRule.resolvedHost) {
+      continue;
+    }
+
+    const matchedHost = hostRule.matchHost;
+    // Should never be needed as if we have a resolvedHost there has to be a matchHost
+    // istanbul ignore next
+    if (!matchedHost) {
+      continue;
+    }
+
+    const uri = `//${matchedHost}/`;
+    let cleanedUri = uri;
+    if (is.string(matchedHost) && isHttpUrl(matchedHost)) {
```

Comment:
why the additional is.string check?

Reply:
left over from before the falsely check has been added.